### PR TITLE
[Issue 12271][Modernizer] Add Maven Modernizer plugin to pulsar-metadata module

### PR DESCRIPTION
**Pull request by jadireddi**
Master Issue: #12271 

### Motivation

Apply Maven Modernizer plugin to detect uses of legacy APIs which can be replaced with modern APIs that are often more performant, safer, and idiomatic than the legacy equivalents

### Modifications

Add Maven Modernizer plugin to `pulsar-metadata` module and fix the violations.

### Verifying this change

- [ yes] Make sure that the change passes the CI checks.

This change is a code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (no)
  - The public API: (no)
  - The schema: (no)
  - The default values of configurations: (no)
  - The wire protocol: (no)
  - The rest endpoints: (no)
  - The admin cli options: (no)
  - Anything that affects deployment: ( no)

### Documentation

Need to update docs? 
  
- [x] `no-need-doc`

### CODE REVIEW

**Comment by jadireddi** [comment]
PR is ready for review. As I am a first-time contributor, can you please approve the workflows.

**Review by MarvinCai — APPROVED** [review]
lgtm

**Review by codelipenghui — APPROVED** [review]

**Comment by codelipenghui** [comment]
@jadireddi Could you please rebase your branch? #12270 already introduced `maven-modernizer-plugin.version` in the pom.xml.

**Comment by jadireddi** [comment]
@codelipenghui , Rebased on the latest master. Could you please review and approve the workflow to run, as it's my first PR.

**Review by nicoloboschi — APPROVED** [review]

File: pulsar-metadata/pom.xml [hunk 1]
```diff
@@ -94,6 +94,23 @@
           </execution>
         </executions>
       </plugin>
+      <plugin>
+        <groupId>org.gaul</groupId>
+        <artifactId>modernizer-maven-plugin</artifactId>
+        <version>${maven-modernizer-plugin.version}</version>
+        <configuration>
```
Comment (nicoloboschi) [inline pulsar-metadata/pom.xml:101]: is it better to set the configuration in the root POM and share it with all submodules

cc @MarvinCai

**Review by nicoloboschi — CHANGES_REQUESTED** [review]

File: pulsar-metadata/pom.xml (hunk 1 above)
Comment (MarvinCai) [reply pulsar-metadata/pom.xml:101]: yeah, we do that in mainline branch, this one might need a rebase.

**Comment by MarvinCai** [comment]
@jadireddi can you try rebase from current mainline and remove plugin version from submodule? thanks

File: pulsar-metadata/pom.xml (hunk 1 above)
Comment (jadireddi) [reply pulsar-metadata/pom.xml]: Fixed review comments. Could you please review.

**Comment by MarvinCai** [comment]
@jadireddi we just need to remove plugin version from submodule, not the whole plugin usage.

**Comment by jadireddi** [comment]
@MarvinCai,  Fixed. Sorry for the confusion.

**Comment by MarvinCai** [comment]
from test result seems there're some additional violations from mainline that we need to take care

**Comment by jadireddi** [comment]
Could you please approve the workflow to kick in

**Comment by codelipenghui** [comment]
@jadireddi Sorry for the late response, the CI is running for now.

**Pull request merged**